pkg/prometheus/statefulset.go

```diff
@@ -542,6 +542,66 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *Config) (*appsv1.Stateful
 
 	finalLabels := c.Labels.Merge(podLabels)
 
+	additionalContainers := p.Spec.Containers
+
+	if p.Spec.Thanos != nil {
+		thanosBaseImage := c.ThanosDefaultBaseImage
+		if p.Spec.Thanos.BaseImage != nil {
+			thanosBaseImage = *p.Spec.Thanos.BaseImage
+		}
+
+		thanosArgs := []string{
+			"sidecar",
+			fmt.Sprintf("--cluster.peers=%s", p.Spec.Thanos.Peers),
```

Comment:
Also, the `--cluster.peers` should not be required. In my use-case, the thanos-sidecar is alone and connected by a remote thanos-query --store parameter.

Reply:
Got it. Thanks for the insight! I've only run with both.

Reply:
sidecar will never block. It will just print this warning until it will be not alone. Also, we use gossip by default here in operator (since it is scoped to single cluster only for now), so having a mandatory cluster.peers seems fine.

Reply:
@Bplotka I had the issue. After our discussion on Slack I did the modifications you recommended and my thanos-query couldn't communicate with the thanos-sidecar (`context deadline exceeded`) when the thanos-sidecar was spamming "I appear to be alone in the cluster". I have then removed the `--cluster.peers`, restart the prometheus+thanos-sidecar and it worked.

Reply:
For now I'm marking `Peers` within the `Thanos` config as optional, the entire integration is marked experimental for now so we can still make modifications before we promote it to be stable.